app/services/workflow_step_service.rb

```diff
@@ -8,7 +8,19 @@ def initialize(course)
   end
 
   def call
-    if course.is_further_education?
+    if course.recruitment_cycle_after_2023? # This can be removed after the 2023 cycle has ended
+      if course.is_further_education?
+        next_cycle_further_education_workflow_steps - remove_study_site_if_feature_flag_disabled_or_current_cycle(course)
+      elsif course.is_uni_or_scitt?
+        next_cycle_uni_or_scitt_workflow_steps - visas_to_remove(course) - remove_study_site_if_feature_flag_disabled_or_current_cycle(course)
+      elsif course.is_school_direct?
+        if course.provider.accredited_bodies.length == 1
+          school_direct_workflow_steps - (visas_to_remove(course) + [:accredited_provider]) - remove_study_site_if_feature_flag_disabled_or_current_cycle(course)
+        else
+          school_direct_workflow_steps - remove_study_site_if_feature_flag_disabled_or_current_cycle(course) - visas_to_remove(course)
+        end
+      end
+    elsif course.is_further_education?
```

Comment:
For the first time reading the code, each branches of the if returns a workflow with the steps. Is it possible to move each to your own method like:

```
if course.is_further_education?
  further_education_steps
elsif course.is_uni_or_scitt?
  uni_or_scitt_steps # or uni_or_scitt_workflow (up to you)
...
```

Reply:
I think this will make the code cleaner, wdyt?

Reply:
I've done that here https://github.com/DFE-Digital/publish-teacher-training/pull/3683/commits/563fc220615c5f79825c53d677286cf5092addb6

But this is an area that we should visit to refactor in the future because there's not much difference between the steps and after rollover a lot of the complexity will be removed.